docs/getting_started.rst

```diff
@@ -31,105 +39,467 @@ example to construct such a store for the local filesystem.
 
 .. ipython:: python
 
-   from storefact import get_store_from_url
-   from tempfile import TemporaryDirectory
+    from functools import partial
+    from storefact import get_store_from_url
+    from tempfile import TemporaryDirectory
+
+    dataset_dir = TemporaryDirectory()
 
-   dataset_dir = TemporaryDirectory()
-   store = get_store_from_url(f"hfs://{dataset_dir.name}")
+    store_factory = partial(get_store_from_url, f"hfs://{dataset_dir.name}")
 
-Now that we have our data and the storage location, we can persist it to a dataset.
-For that we use in this guide :func:`kartothek.io.eager.store_dataframes_as_dataset`
-to store a ``DataFrame`` we already have in memory in the local task.
+The reason ``store_factory`` is defined as a ``partial`` callable with the store
+information as arguments is because,
+when using distributed computing backends in ``kartothek``,
+the connections of the store cannot be safely transferred between processes,
+thus, we pass storage information to workers as a factory function.
 
-The import path of this function already gives us a hint about the general
-structuring of the ``kartothek`` modules. In :mod:`kartothek.io` we have all
-the building blocks to build data pipelines that read and write from/to storages.
-Other top-level modules for example handle the serialization of DataFrames to
-``bytes``.
 
-The next module level ``eager`` describes the scheduling backend.
-``eager`` runs all execution immediately and on the local machine.
-There is also ``iter`` that supports reading the dataset on
-a per-partition base. For larger dataset, ``dask`` can be used
-to work on datasets in parallel or even in a cluster by using
-``distributed`` as the backend for ``dask``.
+Writing dataset to storage
+===========================
+
+Now that we have our data and the storage location, we can persist the dataset.
+To do so, we will use :func:`kartothek.io.eager.store_dataframes_as_dataset`
+to store a ``DataFrame`` we already have in memory.
```

Comment:
> Ideally I'd hope that a user can execute the statements one by one when opening an IPython shell
Yes, it has to be so because the ipython directive within the doc itself persists the session across cells, so building the docs would throw an error if that weren't the case.